src/components/poolstats/PoolPositionWidget.tsx

```diff
@@ -39,10 +49,175 @@ const PerformanceCard = styled.div`
 
 export type PoolPositionWidgetProps = {
   poolData: BlendPoolMarkers;
+  offChainPoolStats: OffChainPoolStats | undefined;
 };
 
+type AccountStats = {
+  accountValue: number,
+  accountValue1DAgo: number | null,
+  todaysReturn: number | null,
+  totalReturn: number | null,
+};
+
+type Timestamp = {
+  timestamp: string | number;
+}
+
+function makeEtherscanRequest(
+  fromBlock: number,
+  address: string,
+  topics: (string | null)[],
+  shouldMatchAll: boolean,
+  pageLength = 1000,
+  page?: number,
+  toBlock?: number
+) {
+  let query = `https://api.etherscan.io/api?module=logs&action=getLogs`.concat(
+    `&fromBlock=${fromBlock.toFixed(0)}`,
+    toBlock ? `&toBlock=${toBlock.toFixed(0)}` : '',
+    `&address=${address}`
+  );
+  
+  for (let i = 0; i < topics.length; i += 1) {
+    if (topics[i] === null) continue;
+    query += `&topic${i}=${topics[i]}`;
+
+    if (i === topics.length - 1) break;
+    query += `&topic${i}_${i + 1}_opr=${shouldMatchAll ? 'and' : 'or'}`;
+  }
+
+  if (page) query += `&page=${page}`;
+  query = query.concat(`&offset=${pageLength}`, `&apikey=${process.env.REACT_APP_ETHERSCAN_API_KEY}`);
+
+  return axios.get(query);
+}
+
+function findNearestElementByTime<T extends Timestamp>(arr: T[], timestamp: string | number): T {
```

Comment:
This is fine for now, but I believe I have a better way to do this that we can look into in the future